ingesters/namedpipe/namedpipeingester.go

```diff
@@ -0,0 +1,77 @@
+package namedpipe
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"os"
+
+	"go.uber.org/zap"
+
+	"github.com/metal-toolbox/audito-maldito/internal/health"
+)
+
+const (
+	// NamedPipeProcessorComponentName is the name of the component
+	// that reads from a named pipe. This is used in the health check.
+	NamedPipeProcessorComponentName = "named-pipe-processor"
+)
+
+type NamedPipeIngester struct{}
+
+type Callback func(context.Context, string) error
+
+func (n *NamedPipeIngester) Ingest(
+	ctx context.Context,
+	filePath string,
+	delim byte,
+	callback Callback,
+	logger *zap.SugaredLogger,
+	h *health.Health,
```

Comment:
It might be a good idea to move the logger and health to the `NamedPipeIngester` structure itself.

Reply:
will do.